# **Importing Libraries**

In [11]:
#Importing required libraries
import pandas as pd
import numpy as np


# **Reading Files**

In [12]:
#Reading csv files
df1=pd.read_csv('/content/olist_customers_dataset.csv.zip')
df2=pd.read_csv('/content/olist_order_items_dataset.csv.zip')
df3=pd.read_csv('/content/olist_order_payments_dataset.csv.zip')
df4=pd.read_csv('/content/olist_order_reviews_dataset.csv.zip')
df5=pd.read_csv('/content/olist_orders_dataset.csv.zip')
df6=pd.read_csv('/content/olist_products_dataset.csv.zip')
df7=pd.read_csv('/content/olist_sellers_dataset.csv')
df8=pd.read_csv('/content/product_category_name_translation.csv')

In [13]:
#Columns in different dataframes
for name, df in zip(['customers_dataset','order_items_dataset', 'order_payments_dataset', 'order_reviews_dataset', 'orders_dataset.csv', 'products_dataset', 'sellers_dataset', "product_category_name_translation"], [df1, df2, df3, df4, df5, df6, df7, df8]):
    print(f"--- columns from {name} ---")
    display(df.columns)
    print("\n")

--- columns from customers_dataset ---


Index(['customer_id', 'customer_unique_id', 'customer_zip_code_prefix',
       'customer_city', 'customer_state'],
      dtype='object')



--- columns from order_items_dataset ---


Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='object')



--- columns from order_payments_dataset ---


Index(['order_id', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value'],
      dtype='object')



--- columns from order_reviews_dataset ---


Index(['review_id', 'order_id', 'review_score', 'review_comment_title',
       'review_comment_message', 'review_creation_date',
       'review_answer_timestamp'],
      dtype='object')



--- columns from orders_dataset.csv ---


Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'],
      dtype='object')



--- columns from products_dataset ---


Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')



--- columns from sellers_dataset ---


Index(['seller_id', 'seller_zip_code_prefix', 'seller_city', 'seller_state'], dtype='object')



--- columns from product_category_name_translation ---


Index(['product_category_name', 'product_category_name_english'], dtype='object')

In [14]:
#Displaying different csv files
for name, df in zip(['customers_dataset', 'order_items_dataset', 'order_payments_dataset', 'order_reviews_dataset', 'orders_dataset.csv', 'products_dataset', 'sellers_dataset', "product_category_name_translation"], [df1, df2, df3, df4, df5, df6, df7, df8]):
    print(f"--- Sample from {name} ---")
    display(df.sample(5))
    print("\n")

--- Sample from customers_dataset ---


,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
40365,796ce94893586f6db66268eab548a442,f4385a17e1f99f33340a9c51a0e042f3,29702,colatina,ES
42609,3c10da268c72de9078aff090328e7e4e,d2d64464f9572b5caf7af0d7a8f7ef5d,22290,rio de janeiro,RJ
64582,6cd650d55491e2da73465ce4a420afde,cd4e980e57f18e298b12e0c7bb11de23,80230,curitiba,PR
70290,73d3109d331fc1a1ab21f489b3979ef8,adc7533e622049e2314ed25c6df7e95d,89802,chapeco,SC
80350,37000e0d03f1d8fccec6cac3a8020850,8bb887dc30e417bbe7a31025a3eef3ab,13603,araras,SP




--- Sample from order_items_dataset ---


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
31062,4677f9438ab0c129fd625699019e160f,1,31c79131e883e5fd8c4c85fe9f7d2bb2,ea8482cd71df3c1969d7b9473ff13abc,2018-01-05 13:46:46,24.99,7.78
91126,cee43113ca0a71b50e899098ec4c5911,1,750dcfda0a9788bba89c94bb82a322db,53e4c6e0f4312d4d2107a8c9cddf45cd,2018-04-12 18:30:19,22.99,11.15
37310,54a845f5cc0d3691277e4adfebd4a5bf,1,c354569eefd558bf201d0aedf35608ed,822166ed1e47908f7cfb49946d03c726,2017-09-20 02:55:21,49.90,16.11
33737,4c5fb984844ad48dc34f8acf42eb0eda,1,331e79a28ee5afbb321a3146653085f2,7c67e1448b00f6e969d365cea6b010ab,2017-04-06 17:02:14,176.99,60.82
9196,15044169648e6a6e9b1ffe21405b6391,1,b114bf337c0626166abe574eee9e3f32,7c67e1448b00f6e969d365cea6b010ab,2017-12-15 04:40:06,149.94,36.71




--- Sample from order_payments_dataset ---


,order_id,payment_sequential,payment_type,payment_installments,payment_value
30782,0e7a11a73554ebd7fa1de70d50b75285,1,credit_card,1,27.53
82750,082730bdb06cc2b755fbb33becd6ccec,1,credit_card,1,48.31
52350,d98b754dc197354be163fe4f75ef28fc,1,credit_card,8,123.68
100062,9099fbdbf8ce8489a7546442a0881f93,2,voucher,1,213.89
49457,d19792edee7b2cc1d094d869461abf3a,1,boleto,1,131.09




--- Sample from order_reviews_dataset ---


,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
46307,99c83e1d9eb3a23d6369e51c1d9f2dc9,dc2249cafaf7687c5b9d86fd2a3198eb,5,NaN,NaN,2018-04-07 00:00:00,2018-04-09 15:44:38
20392,99b3d4b8f7d9d2fce5477fcee94e6f37,eff124e4c7161cdbc3ecd912e716715e,4,NaN,NaN,2017-10-05 00:00:00,2017-10-06 03:15:14
83164,8b6456226e6dd0691635926026ca18c5,9c37d7a7de99233fd6490a1c09ba0404,3,recomendo,"Entregou dentro do prazo, mas o tamanho dos \r...",2018-07-14 00:00:00,2018-07-16 23:17:36
58622,dd7ad46e529dee82ac30a6f2561e0aa3,159b0ce7faa02427505f3ce47f48e933,5,recomendo,Ótima loja e produtos,2018-08-29 00:00:00,2018-08-29 11:31:04
91012,33af0eac076a9d5412f4e5988e824e4e,389b5ea850fb0283bb94a04dca7750ec,4,NaN,NaN,2017-03-23 00:00:00,2017-03-24 00:08:23




--- Sample from orders_dataset.csv ---


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
59861,a0b5c0a93889a160e7f4638322b9959c,6104e28eeabc55665817b8897914fbc9,delivered,2018-08-15 12:25:40,2018-08-15 12:44:15,2018-08-15 14:22:00,2018-08-21 10:48:41,2018-09-03 00:00:00
99432,cfa78b997e329a5295b4ee6972c02979,a2f7428f0cafbc8e59f20e1444b67315,delivered,2017-12-20 09:52:41,2017-12-20 10:09:52,2017-12-20 20:25:25,2018-01-26 15:45:14,2018-01-18 00:00:00
32853,c1a3f085dae6065e1cdff2df7d95d990,f961764b9600b0fd7ba8e43a21daae4d,delivered,2017-05-09 13:21:53,2017-05-09 13:31:22,2017-05-17 08:32:17,2017-05-25 10:13:56,2017-05-25 00:00:00
55120,73fa09600b87d64333b03f2fe849e096,1fbb2a549b8f58e925ad931401717418,delivered,2017-09-12 19:15:03,2017-09-12 19:25:16,2017-09-14 17:07:56,2017-09-18 21:27:43,2017-10-09 00:00:00
57750,ab5ff8523a162735e1d5ce37a2e4be76,beda2760f494f9ad8fafa5343c4d9a85,delivered,2018-06-22 09:02:41,2018-06-23 02:20:36,2018-06-27 08:59:00,2018-07-05 17:28:52,2018-07-19 00:00:00




--- Sample from products_dataset ---


,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
9847,c1eee16d856657ad280936810ef3d080,cool_stuff,59.0,1758.0,1.0,113.0,22.0,8.0,17.0
1490,56a5da29b4975931b470aa9d61ced8d3,automotivo,52.0,1236.0,1.0,200.0,20.0,6.0,17.0
25722,d3796e73f7d260e45f5d5b9dc1c5a7a1,cama_mesa_banho,38.0,567.0,1.0,2600.0,37.0,23.0,35.0
1573,bd9d35f2bce5b870f79e5d7112a01953,beleza_saude,54.0,457.0,4.0,1700.0,36.0,10.0,36.0
1522,eed6978726f666f29319519d56058469,perfumaria,42.0,3923.0,1.0,150.0,16.0,8.0,11.0




--- Sample from sellers_dataset ---


,seller_id,seller_zip_code_prefix,seller_city,seller_state
1993,d32e3909cfb714fb2a6cb092c85b9c5f,13215,jundiai,SP
832,ff4ea69c2a729e83e63c7579e4ef8170,7071,guarulhos,SP
2333,77128dec4bec4878c37ab7d6169d6f26,2610,sao paulo,SP
458,89d9a386b0b6e5fc8403071b03f7206a,3103,sao paulo,SP
2107,cca3071e3e9bb7d12640c9fbe2301306,14940,ibitinga,SP




--- Sample from product_category_name_translation ---


,product_category_name,product_category_name_english
53,pcs,computers
34,artigos_de_festas,party_supplies
49,construcao_ferramentas_seguranca,construction_tools_safety
51,fashion_esporte,fashion_sport
21,fashion_calcados,fashion_shoes


# **Merging files**

In [15]:
#Merging all columns of all files(except Geolocation file)
orders_customers = pd.merge(df5, df1, on='customer_id', how='left')
order_items_products = pd.merge(df2, df6, on='product_id', how='left')
order_items_products_sellers = pd.merge(order_items_products, df7, on='seller_id', how='left')
orders_items_products_sellers_customers = pd.merge(orders_customers, order_items_products_sellers, on='order_id', how='left')
merged_df = pd.merge(orders_items_products_sellers_customers, df4, on='order_id', how='left')
merged_df = pd.merge(merged_df, df5, on='order_id', how='left')

merged_df = pd.merge(merged_df, df8, on='product_category_name', how='left')

# Geolocation data not merged

display(merged_df.shape)
display(merged_df.columns)
display(merged_df.sample(5))

(114092, 43)

Index(['order_id', 'customer_id_x', 'order_status_x',
       'order_purchase_timestamp_x', 'order_approved_at_x',
       'order_delivered_carrier_date_x', 'order_delivered_customer_date_x',
       'order_estimated_delivery_date_x', 'customer_unique_id',
       'customer_zip_code_prefix', 'customer_city', 'customer_state',
       'order_item_id', 'product_id', 'seller_id', 'shipping_limit_date',
       'price', 'freight_value', 'product_category_name',
       'product_name_lenght', 'product_description_lenght',
       'product_photos_qty', 'product_weight_g', 'product_length_cm',
       'product_height_cm', 'product_width_cm', 'seller_zip_code_prefix',
       'seller_city', 'seller_state', 'review_id', 'review_score',
       'review_comment_title', 'review_comment_message',
       'review_creation_date', 'review_answer_timestamp', 'customer_id_y',
       'order_status_y', 'order_purchase_timestamp_y', 'order_approved_at_y',
       'order_delivered_carrier_date_y', 'order_delivered_custo

,order_id,customer_id_x,order_status_x,order_purchase_timestamp_x,order_approved_at_x,order_delivered_carrier_date_x,order_delivered_customer_date_x,order_estimated_delivery_date_x,customer_unique_id,customer_zip_code_prefix,...,review_creation_date,review_answer_timestamp,customer_id_y,order_status_y,order_purchase_timestamp_y,order_approved_at_y,order_delivered_carrier_date_y,order_delivered_customer_date_y,order_estimated_delivery_date_y,product_category_name_english
27488,817aec5fee39a2d4968cb6c48bad78f1,616c3d715afc55e0ae24fcfca7780796,delivered,2018-05-16 20:57:35,2018-05-16 21:14:08,2018-05-17 11:22:00,2018-05-22 17:42:11,2018-06-06 00:00:00,786ccad63616e631472911d4762363f5,17015,...,2018-05-23 00:00:00,2018-05-24 12:08:59,616c3d715afc55e0ae24fcfca7780796,delivered,2018-05-16 20:57:35,2018-05-16 21:14:08,2018-05-17 11:22:00,2018-05-22 17:42:11,2018-06-06 00:00:00,perfumery
40863,4985a7fdb01bf57b6546b9ea5790adbd,802e8b67850e74f3d9894d4f5a3035e1,delivered,2018-06-09 16:50:56,2018-06-09 17:10:03,2018-06-11 14:02:00,2018-07-08 18:44:22,2018-07-13 00:00:00,e7ea4c3049b0ffcb326d8fcf44a06c05,40296,...,2018-07-09 00:00:00,2018-07-09 22:07:58,802e8b67850e74f3d9894d4f5a3035e1,delivered,2018-06-09 16:50:56,2018-06-09 17:10:03,2018-06-11 14:02:00,2018-07-08 18:44:22,2018-07-13 00:00:00,health_beauty
61109,eadacd4909367cdc58880cf410cb7c4c,a765d1107969e6412dc25b6529dbf582,delivered,2018-01-17 23:59:58,2018-01-18 00:51:35,2018-01-19 23:33:21,2018-02-06 23:13:29,2018-02-15 00:00:00,79e0e65fc52c0126f696eaa055ad4a0d,42701,...,2018-02-07 00:00:00,2018-02-14 20:50:37,a765d1107969e6412dc25b6529dbf582,delivered,2018-01-17 23:59:58,2018-01-18 00:51:35,2018-01-19 23:33:21,2018-02-06 23:13:29,2018-02-15 00:00:00,furniture_decor
23374,a257a1b730eacbbf17f67bfd45703f8f,50fa790025bcefae879b3c497abf5f33,delivered,2017-02-13 14:48:03,2017-02-13 15:02:50,2017-02-14 10:31:54,2017-02-20 11:27:52,2017-03-10 00:00:00,7d0afa050beda0bcbdbe4f021eddb835,13720,...,2017-02-21 00:00:00,2017-02-22 01:52:52,50fa790025bcefae879b3c497abf5f33,delivered,2017-02-13 14:48:03,2017-02-13 15:02:50,2017-02-14 10:31:54,2017-02-20 11:27:52,2017-03-10 00:00:00,health_beauty
15105,2f126b65184ad1af3f36fcbf07ae078c,dfa2bf31dd8d1c1a393ee8420947334d,delivered,2017-10-08 16:45:51,2017-10-08 16:56:27,2017-10-10 16:50:16,2017-10-11 18:29:14,2017-10-27 00:00:00,8f710a282ee4fff06ee62707d7f18ea3,13469,...,2017-10-12 00:00:00,2017-10-12 15:30:34,dfa2bf31dd8d1c1a393ee8420947334d,delivered,2017-10-08 16:45:51,2017-10-08 16:56:27,2017-10-10 16:50:16,2017-10-11 18:29:14,2017-10-27 00:00:00,health_beauty


In [16]:
#Merging specific columns of different files
order_data = df5.merge(df2, on='order_id') \
                   .merge(df3, on='order_id') \
                   .merge(df1, on='customer_id') \
                   .merge(df4[['order_id', 'review_score']], on='order_id', how='left') \
                   .merge(df6[['product_id', 'product_category_name']], on='product_id', how='left') \
                   .merge(df8, on='product_category_name', how='left')


# Geolocation data not merged

order_data = order_data.drop('product_category_name', axis=1) #dropping Portuguese product name(keeping English name)

#displaying
display(order_data.shape)
display(order_data.columns)
display(order_data.sample(5))

#order details(id,status,purchase date, approved date, delivery dates,product id, shipping date,price,freight value)
#customer details(id,zip_code,city,state),seller(id,) product etails(id,category_name)
#payement details(payment type,installments,vaue,sequential)

(118307, 24)

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date',
       'order_item_id', 'product_id', 'seller_id', 'shipping_limit_date',
       'price', 'freight_value', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value', 'customer_unique_id',
       'customer_zip_code_prefix', 'customer_city', 'customer_state',
       'review_score', 'product_category_name_english'],
      dtype='object')

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,order_item_id,product_id,...,payment_sequential,payment_type,payment_installments,payment_value,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state,review_score,product_category_name_english
27076,f232ff4908be69d9a06a746be78f1390,6b1066ec623d8c60ca4dc0473a272afc,delivered,2017-08-03 14:16:56,2017-08-03 15:45:26,2017-08-07 20:18:41,2017-08-15 13:15:42,2017-09-05 00:00:00,2,368c6c730842d78016ad823897a372db,...,1,credit_card,3,155.14,3cd77f358fcbe4116c33ac9875f27705,97060,santa maria,RS,4.0,garden_tools
37120,97d3b9b0032e8db77dbde7759a33d984,a393c963191397a4370f0e3119068da2,delivered,2018-05-12 17:37:05,2018-05-12 17:56:58,2018-05-15 16:22:00,2018-05-19 11:23:36,2018-06-06 00:00:00,1,736f1b87428f9cfe5f5184c4ac0fbe05,...,1,credit_card,4,73.20,3f9f86a6984d8f632c8bd9ab3a11dcd6,2352,sao paulo,SP,5.0,bed_bath_table
39443,9d6e82d895f71ace96e14e85f57bcdf7,a1d6e13802152289fcc45e318966308d,delivered,2017-11-26 21:13:40,2017-11-26 21:32:54,2017-11-28 16:31:56,2017-12-06 17:48:17,2017-12-26 00:00:00,1,d1c427060a0f73f6b889a5c7c61f2ac4,...,1,credit_card,1,190.37,e7974277eea2ff1acbecda64bcf45b39,23895,seropedica,RJ,4.0,computers_accessories
73839,7d3636e9d015024df69d61abc7bab409,2bdf82cc0845ed8974063bfba3bd08be,delivered,2018-07-28 02:36:12,2018-07-31 03:55:14,2018-07-31 15:40:00,2018-08-03 20:41:49,2018-08-16 00:00:00,1,c01529871fdb3a98bfd7c845f24f6c21,...,1,boleto,1,78.42,de9c15c18724c6b8b335fe5ff485e01f,11086,santos,SP,5.0,sports_leisure
90020,37467cda81e62d78bf86acfaa7d9b030,a9d80d8258270aa25541958f8c3904f8,delivered,2017-11-14 22:18:48,2017-11-14 22:30:24,2017-11-16 17:14:59,2017-11-29 18:48:52,2017-12-13 00:00:00,1,dd4c3b4ab7e001aaf385ea2e68952030,...,1,credit_card,1,45.00,239eca76b80e32a2d564284842d7ba44,95780,montenegro,RS,4.0,baby


In [17]:
#those col names which are not in order_data but there in merged_data
set(merged_df.columns) - set(order_data.columns)

#check if any of these are required columns

{'customer_id_x',
 'customer_id_y',
 'order_approved_at_x',
 'order_approved_at_y',
 'order_delivered_carrier_date_x',
 'order_delivered_carrier_date_y',
 'order_delivered_customer_date_x',
 'order_delivered_customer_date_y',
 'order_estimated_delivery_date_x',
 'order_estimated_delivery_date_y',
 'order_purchase_timestamp_x',
 'order_purchase_timestamp_y',
 'order_status_x',
 'order_status_y',
 'product_category_name',
 'product_description_lenght',
 'product_height_cm',
 'product_length_cm',
 'product_name_lenght',
 'product_photos_qty',
 'product_weight_g',
 'product_width_cm',
 'review_answer_timestamp',
 'review_comment_message',
 'review_comment_title',
 'review_creation_date',
 'review_id',
 'seller_city',
 'seller_state',
 'seller_zip_code_prefix'}

## **EDA**

In [18]:
order_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118307 entries, 0 to 118306
Data columns (total 24 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   order_id                       118307 non-null  object 
 1   customer_id                    118307 non-null  object 
 2   order_status                   118307 non-null  object 
 3   order_purchase_timestamp       118307 non-null  object 
 4   order_approved_at              118292 non-null  object 
 5   order_delivered_carrier_date   117053 non-null  object 
 6   order_delivered_customer_date  115719 non-null  object 
 7   order_estimated_delivery_date  118307 non-null  object 
 8   order_item_id                  118307 non-null  int64  
 9   product_id                     118307 non-null  object 
 10  seller_id                      118307 non-null  object 
 11  shipping_limit_date            118307 non-null  object 
 12  price                         

In [19]:
order_data.isnull().sum()

,0
order_id,0
customer_id,0
order_status,0
order_purchase_timestamp,0
order_approved_at,15
order_delivered_carrier_date,1254
order_delivered_customer_date,2588
order_estimated_delivery_date,0
order_item_id,0
product_id,0


#We need to remove these null rows for required columns, some columns can be skipped altogether

In [20]:
#Converting data type of date col from object to datetime
date_columns = [
    'order_purchase_timestamp',
    'order_approved_at',
    'order_delivered_carrier_date',
    'order_delivered_customer_date',
    'order_estimated_delivery_date',
    'shipping_limit_date',
]

for col in date_columns:
    order_data[col] = pd.to_datetime(order_data[col], errors='coerce')

order_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118307 entries, 0 to 118306
Data columns (total 24 columns):
 #   Column                         Non-Null Count   Dtype         
---  ------                         --------------   -----         
 0   order_id                       118307 non-null  object        
 1   customer_id                    118307 non-null  object        
 2   order_status                   118307 non-null  object        
 3   order_purchase_timestamp       118307 non-null  datetime64[ns]
 4   order_approved_at              118292 non-null  datetime64[ns]
 5   order_delivered_carrier_date   117053 non-null  datetime64[ns]
 6   order_delivered_customer_date  115719 non-null  datetime64[ns]
 7   order_estimated_delivery_date  118307 non-null  datetime64[ns]
 8   order_item_id                  118307 non-null  int64         
 9   product_id                     118307 non-null  object        
 10  seller_id                      118307 non-null  object        
 11  

In [21]:
#differentiating col as categorical, numerical, datetime columns
categorical_col = [col for col in order_data.columns if order_data[col].dtype == 'object']
datetime_col = [col for col in order_data.columns if order_data[col].dtype == 'datetime64[ns]']
numerical_col = [col for col in order_data.columns if order_data[col].dtype != 'object' and order_data[col].dtype != 'datetime64[ns]']

Need to check if some col seems like numerical/categorical but are categorical/numerical

In [22]:
order_data[categorical_col].describe()

,order_id,customer_id,order_status,product_id,seller_id,payment_type,customer_unique_id,customer_city,customer_state,product_category_name_english
count,118307,118307,118307,118307,118307,118307,118307,118307,118307,116573
unique,98665,98665,7,32951,3095,4,95419,4110,27,71
top,895ab968e7bb0d5659d16cd74cd1650c,270c23a11d024a44c896d1894b261a83,delivered,aca2eb7d00ea1a7b8ebd4e68314663af,4a3ca9315b744ce9f8e9374361493884,credit_card,9a736b248f67d166d2fbb006bcb877c3,sao paulo,SP,bed_bath_table
freq,63,63,115720,536,2155,87258,75,18727,49862,11988


In [23]:
order_data[numerical_col].describe()

,order_item_id,price,freight_value,payment_sequential,payment_installments,payment_value,customer_zip_code_prefix,review_score
count,118307.000000,118307.000000,118307.000000,118307.000000,118307.000000,118307.000000,118307.000000,117329.000000
mean,1.196523,120.648522,20.032823,1.094069,2.943317,172.575102,35051.177276,4.031467
std,0.699474,184.111631,15.836814,0.728579,2.777877,267.110727,29824.786247,1.387927
min,1.000000,0.850000,0.000000,1.000000,0.000000,0.000000,1003.000000,1.000000
25%,1.000000,39.900000,13.080000,1.000000,1.000000,60.850000,11310.000000,4.000000
50%,1.000000,74.900000,16.280000,1.000000,2.000000,108.200000,24310.000000,5.000000
75%,1.000000,134.900000,21.180000,1.000000,4.000000,189.260000,58640.000000,5.000000
max,21.000000,6735.000000,409.680000,29.000000,24.000000,13664.080000,99990.000000,5.000000


In [24]:
order_data[datetime_col].describe()

,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,shipping_limit_date
count,118307,118292,117053,115719,118307,118307
mean,2017-12-30 06:48:50.040707584,2017-12-30 18:16:50.277744896,2018-01-03 08:45:19.290586112,2018-01-12 21:11:40.346200576,2018-01-23 03:38:53.045381632,2018-01-05 22:23:29.509885184
min,2016-09-04 21:15:19,2016-10-04 09:43:32,2016-10-08 10:34:01,2016-10-11 13:46:32,2016-10-20 00:00:00,2016-09-19 00:15:34
25%,2017-09-11 15:16:26,2017-09-12 04:25:26,2017-09-14 19:53:45,2017-09-22 22:10:19,2017-10-02 00:00:00,2017-09-18 14:30:36
50%,2018-01-17 22:39:40,2018-01-18 10:24:29.500000,2018-01-23 17:09:46,2018-02-01 10:57:52,2018-02-14 00:00:00,2018-01-25 04:55:43
75%,2018-05-03 16:34:57,2018-05-03 23:59:33,2018-05-07 14:57:00,2018-05-15 00:13:14.500000,2018-05-25 00:00:00,2018-05-10 02:51:41.500000
max,2018-09-03 09:06:57,2018-09-03 17:40:06,2018-09-11 19:48:28,2018-10-17 13:22:46,2018-10-25 00:00:00,2020-04-09 22:35:08


In [25]:
order_data['order_status'].value_counts()

,count
order_status,
delivered,115720
shipped,1255
canceled,570
invoiced,376
processing,376
unavailable,7
approved,3


#Since we have to calculate CLV, cancelled, shipped, unavailable,processing orders are of not much use to us

In [26]:
order_data.drop(order_data[order_data['order_status'].isin(['canceled', 'shipped', 'unavailable', 'processing'])].index, inplace=True)

In [27]:
order_data['order_status'].value_counts()

,count
order_status,
delivered,115720
invoiced,376
approved,3


In [28]:
order_data.isnull().sum()

,0
order_id,0
customer_id,0
order_status,0
order_purchase_timestamp,0
order_approved_at,15
order_delivered_carrier_date,381
order_delivered_customer_date,387
order_estimated_delivery_date,0
order_item_id,0
product_id,0


Earlier, missing values in order_delivered_carrier_date was 1254
& in order_delivered_customer_date was 2588 which have now reduced by significant amount